lib/cucumber/formatter/http_io.rb

```diff
@@ -7,31 +7,42 @@ class << self
         # Returns an IO that will write to a HTTP request's body
         def open(url, https_verify_mode = nil)
           @https_verify_mode = https_verify_mode
-          uri, method, headers = build_uri_and_headers(url)
+          uri, method, headers = build_uri_method_headers(url)
 
-          method_class_name = "#{method[0].upcase}#{method[1..-1].downcase}"
-          @req = Net::HTTP.const_get(method_class_name).new(uri)
-          headers.each do |header, value|
-            @req[header] = value
-          end
-
-          @http = Net::HTTP.new(uri.hostname, uri.port)
-          if uri.scheme == 'https'
-            @http.use_ssl = true
-            @http.verify_mode = https_verify_mode if https_verify_mode
-          end
+          @req = build_request(uri, method, headers)
+          @http = build_client(uri, https_verify_mode)
 
           read_io, write_io = IO.pipe
           @req.body_stream = read_io
 
-          Thread.new do
-            @http.request(@req)
+          class << write_io
+            attr_writer :request_thread
+
+            def start_request(http, req)
+              @req_thread = Thread.new do
+                begin
+                  res = http.request(req)
+                  if res.code.to_i >= 400
+                    raise Exception.new("request to #{req.uri} failed with status #{res.code}")
```

Comment:
Ought this to be StandardError instead of the top-level Exception?

Reply:
That's fixed now @olleolleolle - thanks!